README.md

```diff
@@ -76,6 +89,124 @@ const App = () => (
 )
 ```
 
+## Props
+
+This is the list of props that you need to pass to the component.
+
+### getIcon
+
+> `function(state: object)` | optional, default icon is provided based on state object
+
+This function decides what icon to show based on the current state of the component.
+
+### getMessage
+
+> `function(icon: string, state: object)` | optional, default message is provided based on the icon and state object.
+
+This function decided what message to show based on the icon (returned from getIcon prop) and the current state of the component.
+
+### getUrl
+
+> `function({})` | optional, no useful default
+
+This function is called as soon as the component enters the viewport and is provided with an element from `props.srcSet`.
+
+### height
+
+> `number` | required
+
+This provides the Height of the image in px.
+
+### icons
+
+> `object` | required
+
+This provides a map of the icons. By default, the component uses icons from material design, implemented as React components with the SVG element. You can customize icons
+
+```js
+const icons = {
+  load: DownloadIcon,
+  //...
+}
+```
+
+### loader
+
+> `string` | optional, defaults to 'xhr'
+
+This prop takes one of the 2 options, `xhr` and `image`. Read more about it [here](https://github.com/stereobooster/react-ideal-image/blob/master/introduction.md#cancel-download).
+
+### placeholder
+
+> `object` | required
+
+This takes one of the 2 objects
+
+```js
+// To add a solid color placeholder
+{
+  color: ''
+}
+```
+
+or
+
+```js
+/**
+ * To add a low quality image
+ * [Low Quality Image Placeholder](https://github.com/zouhir/lqip)
+ * [SVG-Based Image Placeholder](https://github.com/technopagan/sqip)
+ * base64 encoded image of low quality
+ */
+{
+  lqip: ''
+}
+```
+
+Read more about it [here](https://github.com/stereobooster/react-ideal-image/blob/master/introduction.md#lqip).
+
+### shouldAutoDownload
+
+> `function({})` | optional, default function is provided which decides based on the device network.
+
+This function decides if image should be downloaded automatically. The default function returns `false` for a `2g` network,
+for a `3g` network it decides based on `props.threshold` and for a `4g` network it returns `true` by default.
+
+### srcSet
+
+> `array[{}]` | required
```

Comment:
`array[srcType]`, where `srcType` is `{width: number, format: string, src:string}`. Only number is required